engine/rpcserver.go

```diff
@@ -3889,3 +3889,98 @@ func (s *RPCServer) UpdateDataHistoryJobPrerequisite(_ context.Context, r *gctrp
 	}
 	return &gctrpc.GenericResponse{Status: status, Data: fmt.Sprintf("Set job '%v' prerequisite job to '%v' and set status to paused", r.Nickname, r.PrerequisiteJobNickname)}, nil
 }
+
+// StateGetAll returns a full snapshot of currency states, whether they are able
+// to be withdrawn, deposited or traded on an exchange.
+func (s *RPCServer) StateGetAll(_ context.Context, r *gctrpc.StateGetAllRequest) (*gctrpc.StateResponse, error) {
+	exch, err := s.GetExchangeByName(r.Exchange)
+	if err != nil {
+		return nil, err
+	}
+
+	sh, err := exch.GetCurrencyStateSnapshot()
+	if err != nil {
+		return nil, err
+	}
+
+	var resp = &gctrpc.StateResponse{}
+	for x := range sh {
```

Comment:
Quite minor, but if you also override the function `GetCurrencyStateSnapshot` to return a singular snappy, then this for loop will also get coverage